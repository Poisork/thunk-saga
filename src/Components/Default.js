import React from 'react'

export const Star = ({selected=false,onClick=f=>f}) =>(
    <div onClick={onClick}>
        {selected ? 'star selected' : 'star'}
    </div>)