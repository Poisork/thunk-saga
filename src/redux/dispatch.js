import {requestDog,requestDogError,requestDogSuccess} from './actions'

export const fetchDog = () => dispatch => {
    dispatch(requestDog());
    return fetch('https://dog.ceo/api/breeds/image/random')
      .then(res => res.json())
      .then(
        data => dispatch(requestDogSuccess(data)),
        err => dispatch(requestDogError())
      );
  };