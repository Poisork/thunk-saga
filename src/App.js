import React from 'react'; 
import './App.css';
import {connect} from 'react-redux'
import {fetchDog} from './redux/actions'
import {Star} from './Components/Default'
function App(props) { 
  return (
    <div>
      <Star />
      <button onClick={() => props.dispatch(fetchDog())}>Show Dog</button>
        {props.loading 
          ? <p>Loading...</p> 
          : props.error
              ? <p>Error, try again</p>
              : <p><img alt='Dog if you fetch him' style={{width: '500px',height: '500px',objectFit: 'cover', borderRadius: '50%'}} src={props.url}/></p>}
    </div>
  );
}

export default connect(store => store)(App);
