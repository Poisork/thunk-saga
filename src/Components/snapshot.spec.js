import {shallow,mount} from 'enzyme'
import {Star} from './Default'
import React from 'react'
import toJSON from 'enzyme-to-json'
import {compose} from 'redux'
import App from '../App'
import {Provider} from 'react-redux'
import store from '../redux/store'

describe('Shallows',() =>{
    const shallowExpect = compose(expect,toJSON,shallow)
    const mountExpect = compose(expect,toJSON,mount)
 

    it('Render correct properties <Star />',() =>{
        shallowExpect(<Star selected={true} />).toMatchSnapshot()
    })
    it('Render correct properties <App />',() =>{
        mountExpect(  
            <Provider store={store}>
                <App/>
            </Provider>
        ).toMatchSnapshot()
    })
})