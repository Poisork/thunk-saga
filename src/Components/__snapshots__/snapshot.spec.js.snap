// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallows Render correct properties <App /> 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      dispatch={[Function]}
      error={false}
      loading={false}
      url=""
    >
      <div>
        <Star>
          <div
            onClick={[Function]}
          >
            star
          </div>
        </Star>
        <button
          onClick={[Function]}
        >
          Show Dog
        </button>
        <p>
          <img
            alt="Dog if you fetch him"
            src=""
            style={
              Object {
                "borderRadius": "50%",
                "height": "500px",
                "objectFit": "cover",
                "width": "500px",
              }
            }
          />
        </p>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;

exports[`Shallows Render correct properties <Star /> 1`] = `
<div
  onClick={[Function]}
>
  star selected
</div>
`;

exports[`SnapShots Render correct properties <App /> 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(App)>
    <App
      dispatch={[Function]}
      error={false}
      loading={false}
      url=""
    >
      <div>
        <Star>
          <div
            onClick={[Function]}
          >
            star
          </div>
        </Star>
        <button
          onClick={[Function]}
        >
          Show Dog
        </button>
        <p>
          <img
            alt="Dog if you fetch him"
            src=""
            style={
              Object {
                "borderRadius": "50%",
                "height": "500px",
                "objectFit": "cover",
                "width": "500px",
              }
            }
          />
        </p>
      </div>
    </App>
  </Connect(App)>
</Provider>
`;

exports[`SnapShots Render correct properties <Star /> 1`] = `
<div
  onClick={[Function]}
>
  star selected
</div>
`;
